packages/contracts-bedrock/src/L1/DelayedVetoable.sol

```diff
@@ -0,0 +1,159 @@
+        _delay = delay_;
+    }
+
+    /// @notice Gets the initiator
+    /// @return Initiator address.
+    function initiator() external handleCallIfNotVetoer returns (address) {
+        return _initiator;
+    }
+
+    //// @notice Queries the vetoer address.
+    /// @return Vetoer address.
+    function vetoer() external handleCallIfNotVetoer returns (address) {
+        return _vetoer;
+    }
+
+    //// @notice Queries the target address.
+    /// @return Target address.
+    function target() external handleCallIfNotVetoer returns (address) {
+        return _target;
+    }
+
+    /// @notice Gets the delay
+    /// @return Delay address.
+    function delay() external handleCallIfNotVetoer returns (uint256) {
+        return _delay;
+    }
+
+    // TODO(maurelian): Remove this? The contract currently cannot handle forwarding ETH and I'm
+    //   not sure the complexity is warranted.
+    //   If we do allow it:
+    //      1. the callHash will need to include the value
+    //      2. forwarding will need to be done by passing the callHash, rather than the unhashed data
+    /// @notice Used when no data is passed to the contract.
+    receive() external payable {
+        _handleCall();
+    }
+
+    /// @notice Used for all calls that pass data to the contract.
+    fallback() external payable {
+        _handleCall();
+    }
+
+    /// @notice Vetoes a call. This method can only be called by the vetoer. If called by another
+    ///         address, execution will be redirected to _handleCall()
+    function veto(bytes memory data) external handleCallIfNotVetoer {
+        bytes32 callHash = keccak256(data);
+
+        delete _queuedAt[callHash];
+        emit Vetoed(callHash, data);
+    }
+
+    /// @notice Receives all calls other than those made by the vetoer.
+    ///         This enables transparent initiation and forwarding of calls to the target and avoids
+    ///         the need for additional layers of abi encoding.
+    function _handleCall() internal {
+        if (_target == address(0)) {
+            revert TargetUnitialized();
+        }
+
+        bytes32 callHash = keccak256(msg.data);
+        if (_queuedAt[callHash] == 0) {
+            if (msg.sender != _initiator) {
+                revert Unauthorized(_initiator, msg.sender);
+            }
+            _queuedAt[callHash] = block.timestamp;
+            emit Initiated(callHash, msg.data);
+        } else if (_queuedAt[callHash] + _delay < block.timestamp) {
+            // Not enough time has passed, so we'll revert.
+            revert ForwardingEarly();
+        } else {
+            // The ability to finalize the call after sufficient time has passed does not require
+            // authorization.
+
+            // Delete the call to prevent replays
+            delete _queuedAt[callHash];
+
+            // Forward the call
+            emit Forwarded(callHash, msg.data);
+            (bool success,) = _target.call(msg.data);
```

Comment:
Should forward value too

Reply:
That breaks the current mechanism of forwarding by resubmitting the data. See the comment on line 97, if you're ok with the trade off we can add value.

Reply:
Thinking about this more, I do think that we're going to need to allow forwarding value.

In order to achieve that, I will make it so that only the `initiator` passes in unencoded data. All other parties (`vetoer` or any account attempting to forward) will identify the transaction they want to interact with by submitting a tuple of `(nonce, callhash, value)`.